### PR TITLE
saucectl: Add settings for artifacts / artifacts download

### PR DESCRIPTION
**Pull request by FriggaHel**
**Thank you for contributing to the `sauce-docs` project!**
**A well described pull request helps maintainers quickly review and merge your change**

Before submitting your PR, please check our [contributing](https://github.com/saucelabs/sauce-docs/blob/master/CONTRIBUTING.MD)
guidelines, applied for this repository. Avoid large PRs, help reviewers by making them as simple 
and short as possible.

### Description

Add doc for artifacts field in saucectl config file.

### Motivation and Context

Not documented.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [ ] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/master/CONTRIBUTING.MD) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/testrunner-toolkit/configuration/common-syntax.md [hunk 1]
```diff
@@ -297,7 +300,76 @@ __Example__:
   mode: "sauce"
 ```
 
-> For all available resolutions please visit [this documentation page](https://wiki.saucelabs.com/display/DOCS/Test+Configuration+Options#TestConfigurationOptions-SauceLabsCustomTestingOptions).
+## `artifacts`
+
+__Description__: Allows you to define configuration relative to artifacts.
```
Comment (sweeneyskirt-sl) [inline docs/testrunner-toolkit/configuration/common-syntax.md:305]:
```suggestion
__Description__: Specifies how to manage test artifacts, such as logs, videos, and screenshots.
```

File: docs/testrunner-toolkit/configuration/common-syntax.md [hunk 2]
```diff
@@ -297,7 +300,76 @@ __Example__:
...
+  download:
+    when: always
+    match:
+      - junit.xml
+    directory: ./artifacts/
+```
+
+### `download`
+
+__Description__: Allows you to define configuration relative to artifacts downloads.
```
Comment (sweeneyskirt-sl) [inline docs/testrunner-toolkit/configuration/common-syntax.md:321]:
```suggestion
__Description__: Specifies the settings related to downloading artifacts from tests run by `saucectl`.
```

File: docs/testrunner-toolkit/configuration/common-syntax.md [hunk 3]
```diff
@@ -297,7 +300,76 @@ __Example__:
...
+  download:
+    when: always
+    match:
+      - junit.xml
+    directory: ./artifacts/
+```
+
+#### `when`
+
+__Description__: Defines when to download the artifacts.
```
Comment (sweeneyskirt-sl) [inline docs/testrunner-toolkit/configuration/common-syntax.md:336]:
```suggestion
__Description__: Specifies when and under what circumstances artifacts will be downloaded.
```

File: docs/testrunner-toolkit/configuration/common-syntax.md [hunk 4]
```diff
@@ -297,7 +300,76 @@ __Example__:
...
+
+__Type__: *string*
+
+__Values__:
+- `always`: Artfiacts will always be downloaded.
+- `never`: Artifacts will never be downloaded.
+- `pass`: Artifacts will be downloaded only if suite pass.
```
Comment (sweeneyskirt-sl) [inline docs/testrunner-toolkit/configuration/common-syntax.md:343]:
```suggestion
- `pass`: Artifacts will be downloaded only if the suite passes.
```

File: docs/testrunner-toolkit/configuration/common-syntax.md [hunk 5]
```diff
@@ -297,7 +300,76 @@ __Example__:
...
+- `fail`: Artifacts will be downloaded only if suite fails.
```
Comment (sweeneyskirt-sl) [inline docs/testrunner-toolkit/configuration/common-syntax.md:344]:
```suggestion
- `fail`: Artifacts will be downloaded only if the suite fails.
```

File: docs/testrunner-toolkit/configuration/common-syntax.md [hunk 6]
```diff
@@ -297,7 +300,76 @@ __Example__:
...
+
+__Example__:
+```yaml
+    when: always
+```
+
+#### `match`
+
+__Description__: Defines the list of name patterns which artifacts must match to be downloaded. Wildcard character (`*`) can be used.
```
Comment (sweeneyskirt-sl) [inline docs/testrunner-toolkit/configuration/common-syntax.md:353]:
```suggestion
__Description__: Allows you to specify particular files or file types to download based on whether they match the name pattern provided. Wildcard character (`*`) is supported.
```

File: docs/testrunner-toolkit/configuration/common-syntax.md [hunk 7]
```diff
@@ -297,7 +300,76 @@ __Example__:
...
+
+__Example__:
+```yaml
+    - junit.xml
+    - *.log
+```
+
+#### `directory`
+
+__Description__: Defines the folder where artifacts will be downloaded. Each suite will have its own subdirectory.
```
Comment (sweeneyskirt-sl) [inline docs/testrunner-toolkit/configuration/common-syntax.md:365]:
```suggestion
__Description__: Specifies the path to the folder location in which artifacts will be downloaded. Each suite will have its own subdirectory.
```

@FriggaHel -- how does this work? this value is a parent folder and saucectl will create child folders for each suite in the test project?

**Review by sweeneyskirt-sl — CHANGES_REQUESTED** [review]

File: docs/testrunner-toolkit/configuration/common-syntax.md (hunk 4 above)
Comment (alexplischke) [inline docs/testrunner-toolkit/configuration/common-syntax.md:343]:
```suggestion
- `pass`: Artifacts will only be downloaded if the suite passes.
```

File: docs/testrunner-toolkit/configuration/common-syntax.md (hunk 5 above)
Comment (alexplischke) [inline docs/testrunner-toolkit/configuration/common-syntax.md:344]:
```suggestion
- `fail`: Artifacts will only be downloaded if the suite fails.
```

File: docs/testrunner-toolkit/configuration/common-syntax.md (hunk 6 above)
Comment (alexplischke) [inline docs/testrunner-toolkit/configuration/common-syntax.md:353]:
```suggestion
__Description__: Defines the list of name patterns which artifacts must match to be downloaded. Supports the wildcard character `*`.
```

File: docs/testrunner-toolkit/configuration/common-syntax.md (hunk 7 above)
Comment (alexplischke) [inline docs/testrunner-toolkit/configuration/common-syntax.md:365]:
```suggestion
__Description__: Defines the folder where artifacts will be downloaded to. Each suite will have its own subdirectory.
```

File: docs/testrunner-toolkit/configuration/common-syntax.md (hunk 1 above)
Comment (alexplischke) [inline docs/testrunner-toolkit/configuration/common-syntax.md:305]:
```suggestion
__Description__: Allows you to configure how artifacts are handled.
```

File: docs/testrunner-toolkit/configuration/common-syntax.md (hunk 2 above)
Comment (alexplischke) [inline docs/testrunner-toolkit/configuration/common-syntax.md:321]:
```suggestion
__Description__: Allows you to configure when and which artifacts are downloaded.
```

**Review by alexplischke — APPROVED** [review]
LGTM overall (I left some suggestions re. wording)

File: docs/testrunner-toolkit/configuration/common-syntax.md (hunk 7 above)
Comment (FriggaHel) [reply docs/testrunner-toolkit/configuration/common-syntax.md:365]: That's exactly this !

**Review by alexplischke — APPROVED** [review]

File: docs/testrunner-toolkit/configuration/common-syntax.md [hunk 8]
```diff
@@ -297,7 +300,76 @@ __Example__:
...
+
+__Example__:
+```yaml
+    - junit.xml
+    - *.log
+```
+
+#### `directory`
+
+__Description__: Specifies the path to the folder location in which artifacts will be downloaded to. Each suite will have its own subdirectory.
```
Comment (sweeneyskirt-sl) [inline docs/testrunner-toolkit/configuration/common-syntax.md:365]:
```suggestion
__Description__: Specifies the path to the folder location in which artifacts will be downloaded. Each suite will have its own subdirectory.
```

**Review by sweeneyskirt-sl — APPROVED** [review]
Sorry -- I had a typo in one of my suggestions. I corrected it and committed. CI tests are re-running now, When they pass, feel free to merge this.

**Pull request merged**